stl/inc/chrono

```diff
@@ -5177,184 +5217,1230 @@ namespace chrono {
+                // takes the locale information by pointer, but the pointer (from _Gettnames) returns a copy.
+                // _Strftime delegates to other functions but eventually (for the C locale) has the %r specifier
+                // rewritten. It checks for the locale by comparing pointers, which do not compare equal as we have
+                // a copy of the pointer instead of the original. Therefore, we replace %r for the C locale
+                // ourselves.
+                if (_Os.getloc() == locale::classic()) {
+                    _Os << _STD put_time(&_Time, _STATICALLY_WIDEN(_CharT, "%I:%M:%S %p"));
+                    return true;
+                }
+
+                return false;
+            case 'j':
+                if constexpr (_Is_specialization_v<_Ty, duration>) {
+                    _Os << _STD abs(_CHRONO duration_cast<days>(_Val).count());
+                    return true;
+                } else if constexpr (is_same_v<_Ty, month_day>) {
+                    if (_Val.month() > February) {
+                        _THROW(format_error("The day of year for a month_day past February is ambiguous."));
+                    }
+                } else if constexpr (is_same_v<_Ty, month_day_last>) {
+                    if (_Val.month() >= February) {
+                        _THROW(format_error("The day of year for a month_day_last other than January is ambiguous"));
+                    }
+                }
+
+                if constexpr (_Has_ok<_Ty>) {
+                    if (!_Val.ok()) {
+                        _THROW(format_error("Cannot print invalid day of year"));
+                    }
+                }
+                return false;
+            case 'q':
+                if constexpr (_Is_specialization_v<_Ty, duration>) {
+                    _Write_unit_suffix<typename _Ty::period>(_Os);
+                }
+                return true;
+            case 'Q':
+                if constexpr (_Is_specialization_v<_Ty, duration>) {
+                    _Os << _STD abs(_Val.count());
+                }
+                return true;
+            case 'm':
+                if (_Has_modifier) {
+                    return false;
+                }
+
+                if (_Month < 10) {
+                    _Os << _CharT{'0'};
+                }
+                _Os << _Month;
+                return true;
+            case 'Y':
+                if (_Has_modifier) {
+                    return false;
+                }
+
+                if (_Year < 0) {
+                    _Os << _CharT{'-'};
+                }
+                _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:04}"), _STD abs(_Year));
+                return true;
+            case 'y':
+                if (_Has_modifier) {
+                    return false;
+                }
+                _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), _Year < 0 ? 100 + (_Year % 100) : _Year % 100);

```

Comment:
Note: The algorithm for negative `_Year` is incorrect.

Reply:
There's a test for this, `assert(format(STR("{:%Y %y%C}"), year{-1912}) == STR("-1912 88-20"))`, so it appears to be doing what was intended. I think the consensus was that `%y` should be the offset from `%C`*100, rather than abs(`%Y`) mod 100.

Reply:
It fails for -100, -200, ...

Reply:
Open mouth, insert foot. Thanks.

Reply:
Thanks for fixing this in #1872 - merged to `chronat2` then applied to this PR.